pkg/kubectl/cmd/run.go

```diff
@@ -165,6 +165,20 @@ func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *c
 		return fmt.Errorf("Invalid image name %q: %v", imageName, reference.ErrReferenceInvalidFormat)
 	}
 
+	outputFormat := cmdutil.GetFlagString(cmd, "output")
+	showLabelsFlag := cmdutil.GetFlagBool(cmd, "show-labels")
+	dryRunFlag := cmdutil.GetDryRunFlag(cmd)
+	if showLabelsFlag {
+		if dryRunFlag {
+			if outputFormat != "" || outputFormat != "wide" {
```

Comment:
Comparing L173 and L177, it seems `dryRunFlag` does not affect anything if `outputFormat != "wide"`?

Reply:
Yes, `wide` flag makes sure print as follows, **created** or **not**:
```
NAME      TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE       SELECTOR     LABELS
nginx3    ClusterIP   10.0.0.55    <none>        5703/TCP   0s        run=nginx3   <none>

```
When no `dry-run` no `output` just print `xxx creared`

show-labels

|output|""|name|wide|yaml...|
|---|---|---|---|----|
||x|x|o|x|

show-labels with dry-run

|output|""|name|wide|yaml...|
|---|---|---|---|----|
||o|x|o|x|

Reply:
sorry, my carelessness!  It should be `if outputFormat != "" && outputFormat != "wide"`